File: lib/core/license/license_generator.dart
/// Client-side license generation is disabled.
/// Activation codes must be minted by the Invify backend
/// (`POST /devices/activations`) which holds LICENSE_HMAC_SECRET.
class LicenseGenerator {
  static Never generate(String businessName, int durationDays, int planIndex, String deviceSuffix) {
    throw UnsupportedError(
      'Client-side license generation is disabled. Request an activation code from the Invify backend.',
    );
  }
}

File: invify-admin/src/utils/licenseGenerator.js
/**
 * License generation has been moved server-side.
 * Clients must NOT embed HMAC signing secrets.
 * Use deviceApi.createActivation() / deviceApi.validateCode().
 */
export const LicenseGenerator = {
  generate: async function () {
    throw new Error(
      'Client-side license generation is disabled. Use POST /devices/activations via deviceApi.createActivation().',
    );
  },
};
